### DISCUSSION TITLE
Decimal Provider Separation

### DISCUSSION THREAD

**Discussion by adesutherland** [adesutherland/CREXX#367] [General]
I think to 

- future proof the code (for example, to use alternative implementations using GPUs and to allow future updates from Mike),
- to address the issue of the interpreter loop being rather bloated (inevitable but something we need to keep an eye on), and 
- given that the performance of decimal is always going to be relatively slow anyway, 
we need to decouple the decimal library a bit.

I would like a separate c file containing each instruction as a function call with its 3 arguments. The interpreter loop can just call the functions instead of having the login in it. This way we can isolate changes and debug more easily—and I think help with people working different aspects. The decimal maths logic will be separated from the threaded interpreter complexity etc.

@Peter-Jacob especially - are you happy with this approach and should I start the structure off?

Adrian

That way we

**Comment by Peter-Jacob** [comment]
If it doesn't distract you from more important things, go for it.

**Reply by adesutherland** [reply]
When would be a good time for me to do it? What are your weekend plans?

A

**Comment by Peter-Jacob** [comment]
I think there is some time over the weekend. Please let me know what your plan is.
